src/root.h

```diff
@@ -45,8 +47,8 @@ class Root : public Object {
     Child_<Theme> theme;
     Child_<Tmp> tmp;
 
-    RootCommands* root_commands;
     Globals globals;
+    std::unique_ptr<RootCommands> root_commands;
```

Comment:
Is there a reason for this to be a unique_ptr instead of a plain member object? It gets initialized at the beginning of the constructor and is not explicitely deleted as well.

Reply:
I don't want to include `rootcommands.h` in `root.h` (which is included in many cpp-files at the moment). So I want to keep the dependencies between include files at a minimum.

Reply:
I see. Maybe that's worth a comment at that point? E.g. "// wrapped in unique_ptr to avoid include"
As rootcommands.h by itself doesn't bring more baggage in form of includes, I wouldn't expect this particular include to matter though.

Reply:
And as I just found it, this is called _pimpl_-idiom https://www.fluentcpp.com/2017/09/22/make-pimpl-using-unique_ptr/

Reply:
Yes I heard about this before, from Qt I guess. Even a comment "// pimpl" would help though. When I see a smart ptr, I assume there must be some other reason to it.

Reply:
Wouldn't `optional` work here as well (and be slightly more correct)?

Beside that, I'm alson in favor of putting a comment here. I'd write `// Using "pimpl" to avoid huge #include`

Reply:
I like your suggestion, however I feel unique_ptr is indeed the better choice here. As the lifetime of the object here is synchronized with the lifetime of its owner. Optional invites you to always check if its there.

Reply:
You mean this as a semantic distinction, right? Because technically, they're both almost the same (esp. with regards to lifetime and needing to check validity).

In any case, I don't mind unique_ptr either. Go with what you prefer. :-)

Reply:
Yes.

Reply:
I've added the comment. I think `optional` is less correct here, because the root commands object here is mandatory (during the lifetime of Root) and not optional.